typegate/src/utils.ts

```diff
@@ -214,3 +225,82 @@ export function collectFieldNames(tg: TypeGraph, typeIdx: number) {
   }
   return { title: typ?.title, fields: [] };
 }
+
+/**
+ * base64 decode and untar at cwd/{dir}
+ */
+export async function uncompress(dir: string, tarb64: string) {
+  const baseDir = path.join(Deno.cwd(), dir);
+  const buffer = base64.decode(tarb64);
+  const streamReader = new Blob([buffer])
+    .stream()
+    .pipeThrough(new DecompressionStream("gzip"))
+    .getReader();
+  const denoReader = conversion.readerFromStreamReader(streamReader);
+
+  const untar = new Untar(denoReader);
+  const entries = [];
+  for await (const entry of untar) {
```

Comment:
Does not untar.extract() do the same operation?

Reply:
`untar.extract()` works like an iterator so it will have the same effect, both requiring a loop if the root has mutliple entries.